File: 이훈일/Programmers/Level2/2개 이하로 다른 비트.java
class Solution {
    public long[] solution(long[] numbers) {
        long[] answer = new long[numbers.length];
        
        for(int i = 0; i < numbers.length; i++) {
            long number = numbers[i];
            
            if(number % 2 == 0) {
                answer[i] = number + 1;
            } else {
                String numBin = Long.toString(number, 2);
                int index = numBin.lastIndexOf("0");
                String result = "";
                
                if(index == -1) {
                    result = "10" + numBin.substring(1, numBin.length());
                } else {
                    result = numBin.substring(0, index) + "10" + numBin.substring(index+2, numBin.length());
                }
                
                answer[i] = Long.parseLong(result, 2);
            }
        }
        
        return answer;
    }
}

File: 조준장/level2/2개 이하로 다른 비트.kt
class Solution {
    fun solution(numbers: LongArray): LongArray = numbers.map { number ->
            if (number.toInt() % 2 == 0) (number+1).toLong()
            else {
                ("0" + number.toString(2)).let{
                    (it.substring(0, it.lastIndexOf('0')) + "10" + it.substring(it.lastIndexOf('0') + 2, it.length)).toLong(2)
                }
            }
        }.toLongArray()
    
}
